backend/frontend/tests/test_add_review_button.py

```diff
@@ -0,0 +1,164 @@
+    ContentItemFactory,
+)
+from playwright.sync_api import expect
+from curriculum_tracking.models import ContentItem, AgileCard
+from .frontend_test_mixin import FrontendTestMixin
+from activity_log.models import LogEntry
+import curriculum_tracking.activity_log_entry_creators as creators
+from guardian.shortcuts import assign_perm
+from core.models import Team
+
+PROGRESS_DETAILS_VIEW = "progress_details"
+
+class TestProgressDetailsAddReviewButton(FrontendTestMixin):
+    SAMPLE_REVIEW_COMMENT = "This is a comment"
+
+    def setUp(self):
+        super().setUp()
+        self.user = UserFactory()
+        self.user.set_password(self.user.email)
+        self.user.save()
+
+        self.user_with_no_permissions = UserFactory()
+        self.user_with_no_permissions.set_password(self.user_with_no_permissions.email)
+        self.user_with_no_permissions.save()
+
+        self.reviewer_by_assignment = UserFactory()
+        self.reviewer_by_assignment.set_password(self.reviewer_by_assignment.email)
+        self.reviewer_by_assignment.save()
+
+        self.reviewer_by_permission = UserFactory()
+        self.reviewer_by_permission.set_password(self.reviewer_by_permission.email)
+        self.reviewer_by_permission.save()
+
+        self.superuser = UserFactory(is_superuser=True)
+        self.superuser.set_password(self.superuser.email)
+        self.superuser.save()
+
+        self.assignee_team = TeamFactory()
+        self.assignee_team.user_set.add(self.user)
+        self.assignee_team.save()
+
+        assign_perm(
+            Team.PERMISSION_MANAGE_CARDS,
+            self.reviewer_by_permission,
+            self.assignee_team,
+        )
+        assign_perm(
+            Team.PERMISSION_MANAGE_CARDS,
+            self.reviewer_by_assignment,
+            self.assignee_team,
+        )
+
+    def make_in_review_project_card(self):
+        self.card = AgileCardFactory(
+            content_item=ContentItemFactory(
+                content_type=ContentItem.PROJECT,
+                project_submission_type=ContentItem.LINK,
+            ),
+            status=AgileCard.IN_REVIEW,
+        )
+        self.card.assignees.set([self.user])
+        self.card.reviewers.add(self.reviewer_by_assignment)
+        self.card.recruit_project.recruit_users.set([self.user])
+
+    def _get_project_progress_details_url(self, project):
+        return self.reverse_url(
+            PROGRESS_DETAILS_VIEW, kwargs={"id": project.id, "content_type": "project"},
+        )
+    
+    def _submit_review(self, status):
+        self.page.get_by_role("button", name="Add Review").click()
+        self.page.wait_for_load_state()
+        self.page.select_option("select#id_status", status)
+        self.page.query_selector("textarea#id_comments").fill(self.SAMPLE_REVIEW_COMMENT)
+        self.page.query_selector("#submit_review_button").click()
+        self.page.wait_for_load_state('networkidle')
+
+    def test_super_user_can_add_review(self):
```

Comment:
Is there a way to speed up these tests?

If you l;ook at the time it takes to run all the tests, it's friggin long. Not absolutely every test of functionality needs to be a full on integration test. WQe can do small, fast, clear unit tests too. 

Look into the "testing pyramid". Try to only write tests that spin up a browser when necessary. 

If unit tests are slow to run then they become less useful.

You can also write tests that render html but that do not fire up a browser.